### PR TITLE
Add bare reference to docs

### PR DESCRIPTION
**Pull request by heyitsvi**
Bare Reference section includes :
- Overview
- API
- Node.js Compat
- Bare Module

Updated SUMMARY and README for the same.

### CODE REVIEW

File: bare-reference/api.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# API
+
```
Comment (subashsn) [inline bare-reference/api.md:2]: Add an intro, like this

https://docs.pears.com/pear-runtime/api

Comment (subashsn) [inline bare-reference/api.md:2]: Add an intro like in Pear API
https://docs.pears.com/pear-runtime/api

File: bare-reference/overview.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+Bare is built on top of [libjs](https://github.com/holepunchto/libjs), which provides low-level bindings to V8, and [libuv](https://github.com/libuv/libuv), which provides an asynchronous I/O event loop. Bare itself only adds a few missing pieces on top to support a wider ecosystem of modules:
+
+1. A module system supporting both CJS and ESM with bidirectional interoperability between the two.
+2. A native addon system supporting both statically and dynamically linked addons.
+3. Light-weight thread support with synchronous joins and shared array buffer support.
+
+Everything else if left to userland modules to be implemented using these primitives, keeping the runtime itself succinct and _bare_.
+
+## API
+Find the Bare API specs [here](./api.md).
```
Comment (subashsn) [inline bare-reference/overview.md]: New line. + Expand a little hire, just a line. on what it allows it you to do

**Review by subashsn — CHANGES_REQUESTED** [review]

File: bare-reference/api.md (hunk 1 above)
Comment (heyitsvi) [reply bare-reference/api.md:2]: Done

File: bare-reference/overview.md (hunk 2 above)
Comment (heyitsvi) [reply bare-reference/overview.md]: Done

**Review by subashsn — APPROVED** [review]

File: bare-reference/bare-modules.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+```json
+{
+  "exports": "./index.js"
+}
+```
+
+#### `"imports"`
+
+##### Subpath imports
```
Comment (rafapaezbas) [inline bare-reference/bare-modules.md]: Is some explanation missing for Subpath imports, Conditional imports and Private imports?

File: bare-reference/bare-modules.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+##### Private imports
+
+## API
+
+#### `Module.constants`
+
+#### `Module.constants.states`
+
+Constant | Description
```
Comment (rafapaezbas) [inline bare-reference/bare-modules.md]: If description column is empty, we should just remove it.

File: bare-reference/bare-modules.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  main,
+  protocol,
+  imports,
+  resolutions,
+  builtins,
+  conditions
+}
+```
+
+#### `module.url`
```
Comment (rafapaezbas) [inline bare-reference/bare-modules.md]: Should we add some descriptions for these properties?

File: bare-reference/bare-modules.md (hunk 3 above)
Comment (heyitsvi) [reply bare-reference/bare-modules.md]: yes it's missing right now, content ref : https://github.com/holepunchto/bare-module

File: bare-reference/bare-modules.md (hunk 4 above)
Comment (heyitsvi) [reply bare-reference/bare-modules.md]: We did have a discussion on it, included it for completeness but the entire API section needs to be fleshed out. Should we remove the complete 'API' section for now?

**Pull request merged**